packages/jsii/lib/symbol-id.ts

```diff
@@ -11,41 +39,21 @@ export function symbolIdentifier(
     sym = typeChecker.getAliasedSymbol(sym);
   }
 
-  const inFileNameParts: string[] = [];
-
-  let decl: ts.Node | undefined = sym.declarations?.[0];
-  while (decl && !ts.isSourceFile(decl)) {
-    if (
-      ts.isClassDeclaration(decl) ||
-      ts.isNamespaceExportDeclaration(decl) ||
-      ts.isNamespaceExport(decl) ||
-      ts.isModuleDeclaration(decl) ||
-      ts.isEnumDeclaration(decl) ||
-      ts.isEnumMember(decl) ||
-      ts.isInterfaceDeclaration(decl) ||
-      ts.isMethodDeclaration(decl) ||
-      ts.isMethodSignature(decl) ||
-      ts.isPropertyDeclaration(decl) ||
-      ts.isPropertySignature(decl)
-    ) {
-      const name = ts.getNameOfDeclaration(decl);
-      const declSym = name ? typeChecker.getSymbolAtLocation(name) : undefined;
-      if (declSym) {
-        inFileNameParts.unshift(declSym.name);
-      }
-    }
-    decl = decl.parent;
-  }
-  if (!decl) {
+  const tsName = typeChecker.getFullyQualifiedName(sym);
+  // TypeScript fqn looks like "/path/to/file"[.name.in.file]
+  const groups = /^"([^"]+)"(?:\.(.*))?$/.exec(tsName);
+  if (!groups) {
```

Comment:
What does it mean if the name doesn't match the regex? Should we log a warning?

Reply:
I don't know that we have the mechanisms to log a warning. I am fine ignoring this for now, we have end-to-end tests that will ensure everything works correctly. [Alarm on symptoms, not causes](https://paulbellamy.com/2017/08/symptoms-not-causes)

Reply:
I believe we do (`log4js`). But anyway, this is just a minor point.